Assets/MixedRealityToolkit.Tests/PlayModeTests/FocusProviderTests.cs

```diff
@@ -70,6 +70,50 @@ public IEnumerator TestGazeCursorArticulated()
             Assert.IsTrue(inputSystem.GazeProvider.GazePointer.IsInteractionEnabled, "Gaze cursor should be visible after select command");
         }
 
+        /// <summary>
+        /// Tests gaze cursor visibility when near interaction pointer is active and near grabbable.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator TestGazeCursorWhenHandsNearGrabbable()
+        {
+            var inputSystem = PlayModeTestUtilities.GetInputSystem();
+            TestUtilities.PlayspaceToOriginLookingForward();
+
+            // Create grabbable cube
+            var cube = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            cube.AddComponent<NearInteractionGrabbable>();
+            cube.transform.position = new Vector3(0, 0, 4.72f);
+            yield return null;
+
+            // No hands, default cursor should be visible
+            Assert.IsTrue(CoreServices.InputSystem.GazeProvider.GazeCursor.IsVisible, "Head gaze cursor should be visible");
+
+            // Hand up near grabbable
+            TestHand hand = new TestHand(Handedness.Right);
+            yield return hand.Show(Vector3.forward * 4.05f);
+            yield return null;
+
+            var handInput = inputSystem.DetectedInputSources.Where(x => x.SourceName.Equals("Right Hand")).First();
+            IMixedRealityNearPointer grabPointer = (IMixedRealityNearPointer)handInput.Pointers.Where(x => x.PointerName.Contains("Grab")).First();
```

Comment:
FYI, you can use PointerUtils.GetPointer<GrabPointer>() to simplify this lookup (no need for handInput and the LINQ query)

Reply:
I don't like renaming "IsEyeGazeIsValid" - Eye gaze is also used for other things than input (heatmaps). It's important to know when eye gaze is valid and when not. For example, when a user is blinking, the eye gaze may be invalid, which doesn't mean it's in general not meant to be used as an input.

Reply:
I'm confused -- I thought according to Vanessa IsEyeGazeValid means "Is Eye Gaze Being used to control Input", not "is eye gaze information usable / valid currently". 

Perhaps there could be a variable IsEyeGazeValid which means IsEyeTrackingAvailabe && !IsUserBlinking, and a second variable IsEyeGazeUsedToControlnput.

I think the main concern Vanessa & I have is "IsEyeGazeValid" actually seems to mean "is eye tracking currently being used to controle the GazeProvider in MRTK". At least that's how I understand it.

Reply:
Hm, yes, I understand. If I remember correctly, IsEyeGazeValid is not related to whether it is used as input or not. 

Here is some more documentation:
https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/EyeTracking/EyeTracking_EyeGazeProvider.html

_**IsEyeGazeValid**: Indicates whether the current eye tracking data is valid. It may be invalid due to exceeded timeout (should be robust to the user blinking though) or lack of tracking hardware or permissions. Check out our _**Missing eye calibration notification**_ sample that explains how to detect whether a user is eye calibrated and to show an appropriate notification._

In addition, even if the interaction doesn't use eye tracking as an input, developers may still want to access eye tracking data for attention analysis.

Reply:
It seems like this has changed, since [according to Vanessa](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/7246#discussion_r377613528):

> IsEyeGazeValid needs both eye tracking to be available and eye tracking selected as default gaze source. IsEyeTrackingAvailable is there if user wants to query eye tracking ray, even if head gaze is the default.

Reply:
That's a bug with mingling head and eye gaze together in GazeProvider. It would probably be better to handle them as independent input sources. I don't think that this was thoroughly tested and understood when Kurtis originally set it up. Nevertheless, there is a difference between whether the eye tracking data is valid and whether it's used as an input. There could be scenarios in which I want to use the head gaze as a pointer and still be able to access eye tracking data - and in that case it would be great to understand whether the data is valid or not.

Reply:
Before PR, IsEyeGazeValid and IsEyeTrackingValid were calculated like that:

IsEyeTrackingAvailable  returns true if there was recent eye tracking data received from device
IsEyeGazeValid returns IsEyeTrackingAvailable && UseEyeTracking

Documentation says that IsEyeGazeValid reflects valid eye tracking data, but this is not actiually true, as it returns false if head gaze is selected. IsEyeTrackingAvailable actually reflects valid eye data.